library/ecp.c

```diff
@@ -2566,7 +2638,570 @@ static int ecp_mul_mxz( mbedtls_ecp_group *grp, mbedtls_ecp_point *R,
 }
 
 #endif /* MBEDTLS_ECP_MONTGOMERY_ENABLED */
+#if defined(MBEDTLS_ECP_EDWARDS_ENABLED)
+/* sqrt(-1) aka 2^((p-1)/4) */
+static const unsigned char ed25519_sqrt_m1[] = {
+    0x2B, 0x83, 0x24, 0x80, 0x4F, 0xC1, 0xDF, 0x0B,
+    0x2B, 0x4D, 0x00, 0x99, 0x3D, 0xFB, 0xD7, 0xA7,
+    0x2F, 0x43, 0x18, 0x06, 0xAD, 0x2F, 0xE4, 0x78,
+    0xC4, 0xEE, 0x1B, 0x27, 0x4A, 0x0E, 0xA0, 0xB0,
+};
+
+int mbedtls_ecp_point_encode( const mbedtls_ecp_group *grp, mbedtls_mpi *q, const mbedtls_ecp_point *pt )
+{
+    int ret = MBEDTLS_ERR_MPI_BAD_INPUT_DATA;
+
```

Comment:
As this is valid only for Edwards curves, you might want to add:
```
    if( mbedtls_ecp_get_type( grp ) != MBEDTLS_ECP_TYPE_EDWARDS )
        return( MBEDTLS_ERR_ECP_FEATURE_UNAVAILABLE );

Ditto for decode
```

Reply:
I did
```
        h.s *= -1;
        MBEDTLS_MPI_CHK( mbedtls_ecp_mul( grp, &hA, &h, Q, f_rng, p_rng ) );
        MBEDTLS_MPI_CHK( mbedtls_ecp_add( grp, &sB, &sB, &hA ) );
```
but it fails when `mbedtls_mpi_cmp()`. Am I missing something else?